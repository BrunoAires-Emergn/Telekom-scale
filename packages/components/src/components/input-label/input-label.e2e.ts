import { newE2EPage } from '@stencil/core/testing';

describe('scale-input-label', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<scale-input-label/>');
    const element = await page.find('scale-input-label');
    expect(element).toHaveClass('hydrated');
  });
});
