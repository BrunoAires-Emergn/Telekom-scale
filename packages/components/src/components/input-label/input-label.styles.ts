import { JssStyle } from 'jss';

export const styles: JssStyle = {
  'input-label': {
    fontSize: '1rem',
    fontFamily: 'unset',
    paddingBottom: '0.5rem',
    color: 'unset',
  },
};
