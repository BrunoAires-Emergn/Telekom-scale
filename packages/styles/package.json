{
    "name": "@telements/styles",
    "version": "0.0.1",
    "main": "dist/index.js",
    "scripts": {
        "build": "tsc -b",
        "postbuild": "node ./dist/export-css.js",
        "setup": "yarn"
    },
    "author": "@eeegor <egor.kirpichev@telekom.de>",
    "license": "MIT",
    "devDependencies": {
        "typescript": "^3.4.5"
    },
    "dependencies": {

    }
}
