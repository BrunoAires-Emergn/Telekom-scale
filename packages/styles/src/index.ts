import { button } from './button';
import { alert } from './alert';
import { card } from './card';

const theme = {
	colors: {
		white: '#fff',
		black: '#000',
		blue: '#00f',
		red: '#f00',
	}
}

export {
	theme,
	alert,
	button,
	card,
}
