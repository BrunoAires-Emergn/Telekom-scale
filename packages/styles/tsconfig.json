{
	"extends": "../tsconfig.base",
	"compilerOptions": {
		"rootDir": "./src",
		"outDir": "./dist",
		"types": [
			"node",
			"jest"
		]
	}
}
