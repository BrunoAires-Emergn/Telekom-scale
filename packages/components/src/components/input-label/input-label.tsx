import { Component, Prop, h, Host, Method } from '@stencil/core';
import { CssClassMap } from '../../utils/utils';
import classNames from 'classnames';
import { styles } from './input-label.styles';
import { CssInJs } from '../../utils/css-in-js';
import { StyleSheet } from 'jss';
import Base from '../../utils/base-interface';

@Component({
  tag: 't-input-label',
  shadow: true,
})
export class InputLabel implements Base {
  /** (optional) Input text class */
  @Prop() customClass?: string = '';

  /** (optional) Injected jss styles */
  @Prop() styles?: StyleSheet;
  /** decorator Jss stylesheet */
  @CssInJs('InputLabel', styles) stylesheet: StyleSheet;

  componentWillLoad() {}
  componentWillUpdate() {}

  /** Method: updateStyles()  */
  @Method()
  async updateStyles(newStyle: StyleSheet) {
    this.styles = newStyle;
  }

  render() {
    return (
      <Host>
        <style>{this.stylesheet.toString()}</style>
        <div class={this.getCssClassMap()}>
          <slot />
        </div>
      </Host>
    );
  }

  getCssClassMap(): CssClassMap {
    const { classes } = this.stylesheet;
    return classNames(
      classes['input-label'],
      this.customClass && this.customClass
    );
  }
}
