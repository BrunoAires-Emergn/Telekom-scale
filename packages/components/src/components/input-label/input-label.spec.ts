import { newSpecPage } from '@stencil/core/testing';
import { InputLabel } from './input-label';
import { styles } from './input-label.styles';
import jss from 'jss';

describe('Input Label', () => {
  let element;
  beforeEach(async () => {
    element = new InputLabel();
    element.stylesheet = jss.createStyleSheet(styles as any);
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [InputLabel],
      html: `<t-input-label>Label</t-input-label>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should handle css classes', () => {
    element.customClass = 'custom';
    expect(element.getCssClassMap()).toContain('custom');
  });
});
