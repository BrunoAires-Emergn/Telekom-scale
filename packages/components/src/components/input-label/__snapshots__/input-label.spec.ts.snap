// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Label should match snapshot 1`] = `
<scale-input-label>
  <mock:shadow-root>
    <style>
      .input-label-0-0-2 {
  color: unset;
  font-size: 1rem;
  font-family: unset;
  padding-bottom: 0.5rem;
}
    </style>
    <div class="input-label-0-0-2">
      <slot></slot>
    </div>
  </mock:shadow-root>
  Label
</scale-input-label>
`;
